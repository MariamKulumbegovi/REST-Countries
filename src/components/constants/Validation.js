

export const Email_Min_Value=8
export const Password_Min_Value=7
export const Register_Password_Min_Value=6