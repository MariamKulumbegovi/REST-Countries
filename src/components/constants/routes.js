export const HOME_PATH="/"
export const ASKQUESTION_PATH="/Question"
export const LOGIN_PATH="/login"
export const REGISTER_PATH="/register"
export const COUNTER_PATH="/counter"