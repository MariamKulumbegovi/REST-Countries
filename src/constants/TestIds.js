export const TEST_IDS = {
  emailinput: {
    inputElement: 'inputelement',
    hasError: 'valuelenght_hasError',
    isSuccess: 'success@email.com',
  },
  passwordinput: {
    passinput: 'passinput',
    error: 'error',
    success: 'success',
  },
  confirmPassInput: {
    confirmpass: 'confirmpass',
    error: 'error',
    success: 'success',
  },
};
