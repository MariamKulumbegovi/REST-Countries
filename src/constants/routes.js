export const COUNTRIES_PATH = '/countries';
export const LOGIN_PATH = '/login';
export const REGISTER_PATH = '/register';
export const LIST_PATH = '/list';
