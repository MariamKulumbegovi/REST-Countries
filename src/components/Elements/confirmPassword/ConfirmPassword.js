import { useState, useEffect } from 'react';
import { TEST_IDS } from '../../../constants/TestIds';
import { Register_Password_Min_Value } from '../../../constants/Validation';

export const ConfirmPassword = () => {
  const [password, setPassword] = useState('');
  const [passwordError, setPasswordError] = useState('');

  const [confirmPassword, setConfirmPassword] = useState('');
  const [confirmPasswordError, setConfirmPasswordError] = useState('');

  // let number = /[0-9]/;
  // || !number.test(password)
  // || !number.test(password)

  useEffect(() => {
    if (password && password.length < Register_Password_Min_Value) {
      setPasswordError(true);
    } else if (password && password.length >= Register_Password_Min_Value) {
      setPasswordError(false);
    }
  }, [password]);

  useEffect(() => {
    if (confirmPassword != password) {
      setConfirmPasswordError(true);
    } else {
      setConfirmPasswordError(false);
    }
  }, [confirmPassword]);

  const getClassName = () => {
    let errorClassName = '';
    if (passwordError && password) {
      errorClassName = 'is-danger';
    } else if (!passwordError && password) {
      errorClassName = 'is-success';
    }
    return errorClassName;
  };
  const getClassName2 = () => {
    let errorClassName = '';
    if (confirmPasswordError && confirmPassword) {
      errorClassName = 'is-danger';
    } else if (!confirmPasswordError && confirmPassword) {
      errorClassName = 'is-success';
    }
    return errorClassName;
  };

  return (
    <>
      <p className="control has-icons-left has-icons-right">
        <input
          className={`input ${getClassName()}`}
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          required
          onChange={({ target }) => setPassword(target.value)}
          data-testid={TEST_IDS.confirmPassInput.confirmpass}
        />
        <span className="icon is-small is-left">
          <i className="fas fa-lock"></i>
        </span>
        {!passwordError && password && (
          <span className="icon is-small is-right">
            <i className="fas fa-check"></i>
          </span>
        )}
        {passwordError && password && (
          <span className="icon is-small is-right">
            <i className="fas fa-exclamation-triangle"></i>
          </span>
        )}
      </p>
      <p className="control has-icons-left has-icons-right">
        <input
          className={`input ${getClassName2()}`}
          type="password"
          placeholder="Confirm Password"
          name="confirmPassword"
          value={confirmPassword}
          required
          onChange={({ target }) => setConfirmPassword(target.value)}
        />
        <span className="icon is-small is-left">
          <i className="fas fa-lock"></i>
        </span>
        {!confirmPasswordError && confirmPassword && (
          <span className="icon is-small is-right">
            <i className="fas fa-check"></i>
          </span>
        )}
        {confirmPasswordError && confirmPassword && (
          <span className="icon is-small is-right">
            <i className="fas fa-exclamation-triangle"></i>
          </span>
        )}
      </p>
    </>
  );
};
