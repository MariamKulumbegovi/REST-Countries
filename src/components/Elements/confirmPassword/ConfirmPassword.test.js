import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ConfirmPassword } from './ConfirmPassword';
import { TEST_IDS } from '../../../constants/TestIds';

describe('ConfirmPasswordInput', () => {
  it('should render error state', () => {
    const Component = render(<ConfirmPassword />);
    const confirmpass = Component.findByTestId(
      TEST_IDS.confirmPassInput.confirmpass
    );
    userEvent.type(confirmpass, 'error');
    expect(Component.findByTestId(TEST_IDS.confirmPassInput.error))
      .toBeInTheDocument;
  });
  it('should render success state', () => {
    const Component = render(<ConfirmPassword />);
    const confirmpass = Component.getByTestId(
      TEST_IDS.confirmPassInput.confirmpass
    );
    userEvent.type(confirmpass, 'success');
    expect(Component.findByTestId(TEST_IDS.confirmPassInput.success))
      .toBeInTheDocument;
  });
});

describe('Component snap', () => {
  const Component = render(<ConfirmPassword />);
  expect(Component).toMatchSnapshot();
});
