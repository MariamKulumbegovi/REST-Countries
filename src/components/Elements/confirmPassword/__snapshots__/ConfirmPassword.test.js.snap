// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="control has-icons-left has-icons-right"
      >
        <input
          class="input "
          data-testid="confirmpass"
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
        
        
      </p>
      <p
        class="control has-icons-left has-icons-right"
      >
        <input
          class="input "
          name="confirmPassword"
          placeholder="Confirm Password"
          required=""
          type="password"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
        
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="control has-icons-left has-icons-right"
    >
      <input
        class="input "
        data-testid="confirmpass"
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-lock"
        />
      </span>
      
      
    </p>
    <p
      class="control has-icons-left has-icons-right"
    >
      <input
        class="input "
        name="confirmPassword"
        placeholder="Confirm Password"
        required=""
        type="password"
        value=""
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-lock"
        />
      </span>
      
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
